### PR TITLE
Separate building and deploying bundle and index images for CatalogSo…

### PR DESCRIPTION
**Pull request by mgencur**
…urce

https://issues.redhat.com/browse/SRVCOM-1861

Changes:
* build bundle and index images when running in CI or when using DOCKER_REPO_OVERRIDE (then the command would be `DOCKER_REPO_OVERRIDE=docker.io/mgencur make images dev` for installing the CatalogSource
* when not running in CI or with DOCKER_REPO_OVERRIDE use the latest nightly build of the index and bundle images (this also allows for using the `make dev` target from other repositories when the goal is simply to install the latest nightly build of serverless)
* use the new file-based catalogs as described in [OpenShift docs](https://docs.openshift.com/container-platform/4.10/operators/admin/olm-managing-custom-catalogs.html#olm-managing-custom-catalogs-fb) - this format should be fully functional since >=OCP 4.6

For deploying CatalogSource with the latest nightly build we still need to add some configuration to openshift/release to let CI build and promote the index image. The PR is available at https://github.com/openshift/release/pull/28552

### CODE REVIEW

**Comment by mgencur** [comment]
/hold
I will prepare PR for openshift/release to have everything ready when we merge this.

**Comment by matzew** [comment]
/hold

Let's please wait for the `KafkaChannel` migration PR

**Comment by matzew** [comment]
/unhold

(I just see that has been merged)

File: olm-catalog/serverless-operator/index/Dockerfile
Comment (skonto) [inline olm-catalog/serverless-operator/index/Dockerfile:1]: Why not just start with `FROM quay.io/operator-framework/opm:latest`? 
This is what we end-up anyway.

File: hack/lib/catalogsource.bash
Comment (skonto) [inline hack/lib/catalogsource.bash:122]: I guess removing commands such as `opm registry` also makes us [compatible](https://github.com/redhat-openshift-ecosystem/community-operators-prod/discussions/512) with FBC index.

File: hack/lib/catalogsource.bash
Comment (pierDipi) [inline hack/lib/catalogsource.bash:119]: we are using this approach because it's not possible to inject the previously built images using CI operator in a different image build context, right?

**Review by pierDipi — COMMENTED** [review]
Thanks!

File: hack/lib/catalogsource.bash
Comment (mgencur) [reply hack/lib/catalogsource.bash:122]: Exactly. The `registry` command is for the old format. We're now using `init` and `render` to generate the new format.

File: hack/lib/catalogsource.bash
Comment (mgencur) [reply hack/lib/catalogsource.bash:119]: Correct, I did not find a way to get the exact pullspec for the image so that it could be used in the following image build. The pull spec is only available in the "test" containers when running tests, through `dependencies` section of the ci-operator config. And did not get any answer for this on #forum-testplatform: https://coreos.slack.com/archives/CBN38N3MW/p1652275831765409

**Review by pierDipi — APPROVED** [review]
/lgtm

/hold (unhold once you're ready to merge)

**Pull request merged**
